dbms/src/DataStreams/WindowBlockInputStream.cpp

```diff
@@ -1237,6 +1242,9 @@ void WindowTransformAction::writeOutCurrentRow()
 
 Block WindowTransformAction::tryGetOutputBlock()
 {
+    // first try calculate the result based on current data
+    tryCalculate();
```

Comment:
In `WindowTransformOp::transformImpl`, the function always calls `tryCalculate()` before `tryGetOutputBlock`, maybe we can delete them.

Reply:
Before this pr, `tryCalculate()` will calculate as many blocks as possible, so `tryGetOutputBlock()` can be call multiple times after one call to `tryCalculate()`, after this pr, `tryCalculate()` will calculate at most 1 result block, which means if you don't call `tryCalculate()`, `tryGetOutputBlock()` will return 1 block at the first call, and all the subsequent call of `tryGetOutputBlock()`  will return `{}`, which is meaningless and even worse, in pull model, it may cause wrong results. In order to make things right, we should always call `tryCalculate()` before call `tryGetOutputBlock()`, which also means they are coupled together after this pr, so I change `tryCalculate()` to a private method, and call it inside `tryGetOutputBlock()`

Reply:
> Before this pr, `tryCalculate()` will calculate as many blocks as possible, so `tryGetOutputBlock()` can be call multiple times after one call to `tryCalculate()`, after this pr, `tryCalculate()` will calculate at most 1 result block, which means if you don't call `tryCalculate()`, `tryGetOutputBlock()` will return 1 block at the first call, and all the subsequent call of `tryGetOutputBlock()` will return `{}`, which is meaningless and even worse, in pull model, it may cause wrong results. In order to make things right, we should always call `tryCalculate()` before call `tryGetOutputBlock()`, which also means they are coupled together after this pr, so I change `tryCalculate()` to a private method, and call it inside `tryGetOutputBlock()`

Not delete `tryCalculate` in L1246, I mean that we may delete `tryCalculate` call in L47 and L54 in WindowTransformOp.cpp

Reply:
I don't get your point, I've already deleted all the all of `tryCalculate` in `WindowTransformOp.cpp`